shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -184,6 +184,152 @@ abstract class JSBackend(allowUnresolvedSymbols: Bool) {
+      else Rcd(fields.map(f => (f._1, Fld(f._2.flags, desugarQuote(f._2.value)))))
+    case Bra(rcd, trm) =>
+      if (isQuoted) createASTCall("Bra", desugarQuote(trm) :: Nil)
+      else Bra(rcd, desugarQuote(trm))
+    case Sel(receiver, f @ Var(name)) =>
+      if (isQuoted) createASTCall("Sel", desugarQuote(receiver) :: createASTCall("Var", StrLit(name) :: Nil) :: Nil)
+      else Sel(desugarQuote(receiver), f)
+    case Let(rec, Var(name), value, body) =>
+      val letScope = scope.derive("Let")
+      if (isQuoted) {
+        letScope.declareParameter(name)
+        val freshedName = letScope.declareValue(name, S(false), false, N).runtimeName
+        Let(false, Var(freshedName), createASTCall("freshName", StrLit(name) :: Nil),
+          createASTCall("Let", createASTCall("Var", Var(freshedName) :: Nil) :: desugarQuote(value)
+            :: desugarQuote(body)(letScope, isQuoted, new FreeVars(freeVars.vs ++ (name :: Nil))) :: Nil
+        ))
+      }
+      else Let(rec, Var(name), desugarQuote(value), desugarQuote(body)(letScope, isQuoted, freeVars))
+    case Blk(stmts) =>
+      val blkScope = scope.derive("blk")
+      val res = stmts.map {
+        case t: Term =>
+          desugarQuote(t)(blkScope, isQuoted, freeVars)
+        case s => throw CodeGenError(s"statement $s is not supported in quasiquotes")
+      }
+      if (isQuoted) createASTCall("Blk", res)
+      else Blk(res)
+    case Tup(eles) =>
+      def toVar(b: Bool) = if (b) Var("true") else Var("false")
+      def toVars(flg: FldFlags) = toVar(flg.mut) :: toVar(flg.spec) :: toVar(flg.genGetter) :: Nil
+      if (isQuoted) createASTCall("Tup", eles flatMap {
+        case S(Var(name)) -> Fld(flags, t) =>
+          createASTCall("Var", Var(name) :: Nil) :: createASTCall("Fld", desugarQuote(t) :: toVars(flags)) :: Nil
+        case N -> Fld(flags, t) => createASTCall("Fld", desugarQuote(t) :: toVars(flags)) :: Nil
+      })
+      else Tup(eles.map {
+        case v -> Fld(flags, t) => v -> Fld(flags, desugarQuote(t))
+      })
+    case Subs(arr, idx) =>
+      if (isQuoted) createASTCall("Subs", desugarQuote(arr) :: desugarQuote(idx) :: Nil)
+      else Subs(desugarQuote(arr), desugarQuote(idx))
+    case Asc(trm, ty) =>
+      if (isQuoted) desugarQuote(trm)
+      else Asc(desugarQuote(trm), ty)
+    case With(lhs, rhs @ Rcd(fields)) =>
+      if (isQuoted) createASTCall("With", desugarQuote(lhs) :: desugarQuote(rhs) :: Nil)
+      else With(desugarQuote(lhs), Rcd(fields.map(f => (f._1, Fld(f._2.flags, desugarQuote(f._2.value))))))
+    case CaseOf(trm, cases) =>
+      desugarQuotedBranch(cases) match {
+        case L(t) => createASTCall("CaseOf", desugarQuote(trm) :: t :: Nil)
+        case R(b) => CaseOf(desugarQuote(trm), b)
+      }
+    case _ if term.desugaredTerm.isDefined => desugarQuote(term.desugaredTerm.getOrElse(die))
+    case _ => term // * For other terms, either they are not supported & there would be a type error, or we don't need desugar them
```

Comment:
You should list them explicitly here, to make sure you haven't forgotten any. And why not generate a code-gen error instead of silently doing the wrong thing?!

Reply:
You are right. It leads to some problems here. I updated the cases.